clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/operations/BaseOperation.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.client.internal.store.operations;
+
+/**
+ * Base class that represents a {@link org.ehcache.Cache} operation
+ *
+ * @param <K> key type
+ */
+public abstract class BaseOperation<K> {
+
+  public enum Code {
```

Comment:
I agree. All I needed was a simple byte in the `BaseOperation` class to represent the type of operation. But the reason I chose the enum was to clearly map the operation type with a value. If no such mapping exists anywhere, there will be if conditions/switch blocks in the decode methods of codecs that checks for different values before decoding. Code like this:
```java
byte code = buffer.get();
switch(code) {
  case 0:
    //do stuff for PUT
  case 1:
    // do stuff for another operation
  default:
    // something else
}
```

So the mapping will end up defined in the conditional blocks in the decode methods. That is what I was trying to avoid. Define the mapping somewhere centrally. And use it while decoding. And enum provided the added benefit of readability at places where the decoding is done.

So that was the rationale behind the addition of enums. Does it make sense?